nemo_curator/modules/semdedup/extract_dedup_data.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+import yaml
+from tqdm import tqdm
+
+from nemo_curator.modules.semdedup.utils import get_logger
+
+
+def extract_pruned_data(
+    id_col,
+    id_type,
+    sorted_clusters_path,
+    semdedup_pruning_tables_path,
+    eps,
+    num_clusters,
+    output_csv_path,
+):
+
+    ## -- duplicates we want to keep/remove.
+    dedup_clusters = []
+    total = 0
+
+    for cluster_id in tqdm(range(num_clusters)):
+
+        sorted_fname = os.path.join(sorted_clusters_path, f"cluster_{cluster_id}.npy")
+        if not os.path.exists(sorted_fname):
+            logger.info(f"{sorted_fname} not exist. Continue.")
+            continue
+        cluster_i = np.load(sorted_fname)
+
+        df_cluster_i = pd.DataFrame(
+            {
+                id_col: cluster_i[:, 0],
+                "dist": cluster_i[:, 1],
+                "cluster": cluster_i[:, 2],
+            }
+        )
+        df_cluster_i[id_col] = df_cluster_i[id_col].astype(id_type)
+        df_cluster_i.dist = df_cluster_i.dist.astype("float32")
+        df_cluster_i.cluster = df_cluster_i.cluster.astype("int32")
+        total += df_cluster_i.shape[0]
+        cluster_df_fname = os.path.join(
+            semdedup_pruning_tables_path, f"cluster_{cluster_id}.parquet"
+        )
+        semdedup_pruning_tables = pd.read_parquet(cluster_df_fname)
+
+        if semdedup_pruning_tables.shape[0] == 1:
+            logger.info(
+                f"""cluster_id: {cluster_id},
+                    semdedup_pruning_tables.shape: {semdedup_pruning_tables.shape},
+                    df_cluster_i.shape: {df_cluster_i.shape}"""
+            )
+            continue
+
+        ## -- See which examples to keep in this cluster.
+        ## -- semdedup_pruning_tables contain True values for the examples to be removed.
+
+        items_to_keep = semdedup_pruning_tables[
+            semdedup_pruning_tables[f"eps={eps}"] == False
+        ]["id"].tolist()
+
+        if "indices" in semdedup_pruning_tables.columns:
+            cluster_i = cluster_i[semdedup_pruning_tables["indices"]]
+
```

Comment:
No need to check if indices is in columns
```suggestion
        #if "indices" in semdedup_pruning_tables.columns:
        #    cluster_i = cluster_i[semdedup_pruning_tables["indices"]]
```